src/field_theory/ratfunc.lean

```diff
@@ -1095,6 +1095,40 @@ begin
   { exact algebra_map_ne_zero (denom_ne_zero y) },
 end
 
+lemma nat_degree_well_defined {r₁ r₂ s₁ s₂ : polynomial K} (hr₁ : r₁ ≠ 0) (hs₁ : s₁ ≠ 0)
```

Comment:
I am not sure we use `well_defined` in names (indeed it doesn't show up if I look for it). Maybe something like `nat_degree_sub_eq_of_prod_eq`?

Moreover, can you move the lemma to a more basic file? You only need very basic facts about the degree (maybe with a comment saying that this is useful for rational functions).

One last thing: this is maybe annoying , but it would be nice to have a `def` of `nat_degree` (or a similar name, since it has value in `Z`) of a rational function, with a basic API. I guess this can make some of the lemmas below easier to state.

Reply:
I can rename the lemma and move it to some file about polynomials.

Reply:
Yes, I agree that it would be nice to define `degree` for a rational function, I'll do that.

Reply:
`int_degree` seems a good name. I would leave `degree` for another PR (the one taking `bot` at `0`, I don't even know if it's a good definition for rational functions).

Reply:
Ah I see, you're suggesting to define `int_degree` with values in `Z` (and set `int_degree 0 = 0` by convention, I guess).

Reply:
Yes, something like that. For polynomials we have two degree, `nat_degree`, with values in `ℕ`, that sends `0` to `0`, and `degree`, with values in `with_bot ℕ`, that sends `0` to `⊥`, and *a lot* of results connecting the two. It is a little bit annoying, but I am afraid there is no other way. For rational functions there will probably be the same problem, but I would start with the simple one, with values in `ℤ` (the problem is for example that it is not true that the degree of the product is the sum of the degrees, but it is if both rational functions are different from zero of course).

Reply:
I've added `int_degree` and some basic API (a bit more than what I needed to rewrite `infty_valuation` and related lemmas in terms of `int_degree`).